packages/salesforce-adapter/src/metadata_deploy.ts

```diff
@@ -244,6 +243,24 @@ const processDeployResponse = (
     errors.push({ message: result.errorMessage, severity: 'Error' as SeverityLevel })
   }
 
+  const anyErrors = isDefined(result.errorMessage)
+    || componentErrors.length > 0
+    || testErrors.length > 0
+  if (result.rollbackOnError !== false
+    && anyErrors) {
+    // If we deployed with 'rollbackOnError' (the default) and any component in the group fails to deploy, then every
+    // component in the group will not deploy. Let's create an explicit error for the components that did not have
+    // errors to make it clear that they didn't deploy either.
+    makeArray(result.details)
+      .flatMap(detail => makeArray(detail.componentSuccesses))
+      .map(component => ({
+        elemID: typeAndNameToElemId[component.componentType]?.[component.fullName],
+        message: 'This metadata type did not deploy because other types in its deployment group encountered failures',
```

Comment:
@tomermevorach Can you please suggest some text here?
My suggestion:
```suggestion
        message: 'This element has failed to deploy due to other errors in salesforce metadata deployment and rollback on error setting turned on',
```